packages/cosmwasm-stargate/src/signingcosmwasmclient.ts

```diff
@@ -339,21 +339,21 @@ export class SigningCosmWasmClient extends CosmWasmClient {
   public async execute(
     senderAddress: string,
     contractAddress: string,
-    msg: Record<string, unknown>,
+    msgs: Array<Record<string, unknown>>,
     fee: StdFee,
     memo = "",
     funds?: readonly Coin[],
   ): Promise<ExecuteResult> {
-    const executeContractMsg: MsgExecuteContractEncodeObject = {
+    const executeContractMsgs: Array<MsgExecuteContractEncodeObject> = msgs.map((m) => {return {
       typeUrl: "/cosmwasm.wasm.v1.MsgExecuteContract",
       value: MsgExecuteContract.fromPartial({
         sender: senderAddress,
         contract: contractAddress,
-        msg: toUtf8(JSON.stringify(msg)),
+        msg: toUtf8(JSON.stringify(m)),
         funds: [...(funds || [])],
```

Comment:
The funds line here is a problem now. The user would send the funds with every message, so an input of 5 messages and 20 utkn would cost 100 utkn. This is very unintuative.

Reply:
ah got it, will look at this